examples/simple_reverse.coffee

```diff
@@ -0,0 +1,76 @@
+# This is a simple example of how to use the slack-client module in CoffeeScript. It creates a
+# bot that responds to all messages in all channels it is in with a reversed
+# string of the text received.
+#
+# To run, copy your token below, then, from the project root directory:
+#
+# To run the script directly
+#    npm install
+#    node_modules/coffee-script/bin/coffee examples/simple_reverse.coffee 
+#
+# If you want to look at / run / modify the compiled javascript
+#    npm install
+#    node_modules/coffee-script/bin/coffee -c examples/simple_reverse.coffee 
+#    cd examples
+#    node simple_reverse.js
+#
+
+Slack = require '..'
+
+token = 'xoxb-YOUR-TOKEN-HERE' # Add a bot at https://my.slack.com/services/new/bot and copy the token here.
+autoReconnect = true
+autoMark = true
+
+slack = new Slack(token, autoReconnect, autoMark)
+
+slack.on 'open', ->
+  channels = []
+  groups = []
+  unreads = slack.getUnreadCount()
+
+  # Get all the channels that bot is a member of
+  channels = ("##{value.name}" for key, value of slack.channels when slack.channels[key].is_member)
+  # Get all groups that are open and not archived 
+  groups = (for key, value of slack.groups
+    if slack.groups[key].is_open and not slack.groups[key].is_archived
+      value.name)
```

Comment:
I'm going to put it back as a comprehension, then.  I rewrote it slightly to be a bit more obvious what it was doing.